lib/_http_client.js

```diff
@@ -481,8 +491,16 @@ function parserOnIncomingClient(res, shouldKeepAlive) {
     return 2;  // Skip body and treat as Upgrade.
   }
 
-  if (res.statusCode === 100) {
-    // restart the parser, as this is a continue message.
+  // Responses to HEAD requests are crazy.
+  // HEAD responses aren't allowed to have an entity-body
+  // but *can* have a content-length which actually corresponds
+  // to the content-length of the entity-body had the request
+  // been a GET.
+  var isHeadResponse = req.method === 'HEAD';
+  debug('AGENT isHeadResponse', isHeadResponse);
```

Comment:
I removed this a while ago, see https://github.com/nodejs/node/pull/17806#discussion_r158308033.  Can you do the same?

Reply:
Done.

Reply:
It seems like this got back in / was not removed?